File: Servidor/src/main/java/com/gotravel/model/Viaje.java
package com.gotravel.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "viaje")
public class Viaje {

    @Id
    @Column(name = "id_viaje", nullable = false)
    private Integer id;

    @Column(name = "nombre", nullable = false, length = 45)
    private String nombre;

    @Column(name = "descripcion", length = 150)
    private String descripcion;

    @Column(name = "fecha_inicio", nullable = false)
    private LocalDate fechaInicio;

    @Column(name = "fecha_fin", nullable = false)
    private LocalDate fechaFin;

    @Column(name = "coste_total", nullable = false)
    private Double costeTotal;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "id_usuario", nullable = false)
    private Usuario usuario;

}

File: old/ClienteMovil/GoTravel/app/src/main/java/com/gotravel/clienteMovil/data/Viaje.kt
package com.gotravel.clienteMovil.data

import kotlinx.serialization.Serializable
import lombok.NonNull
import lombok.RequiredArgsConstructor

@Serializable
class Viaje (

    val id: Int,
    val nombre: String,
    val descripcion: String?,
    val fechaInicio: String,
    val fechaFin: String?,
    val costeTotal: Double

)